src/Microsoft.Health.Fhir.Liquid.Converter/Filters/SectionFilters.cs

```diff
@@ -96,12 +95,12 @@ public partial class Filters
             return result;
         }
 
-        private static List<object> GetComponents(Hash data)
+        private static List<object> GetComponents(IDictionary<string, object> data)
         {
-            var dataComponents = (((data["ClinicalDocument"] as Hash)?
-                ["component"] as Hash)?
-                ["structuredBody"] as Hash)?
-                ["component"];
+            var clinicalDocument = data["ClinicalDocument"] as IDictionary<string, object>;
+            var component = clinicalDocument?.ContainsKey("component") is true ? clinicalDocument["component"] as IDictionary<string, object> : null;
+            var structuredBody = component?.ContainsKey("structuredBody") is true ? component["structuredBody"] as IDictionary<string, object> : null;
+            var dataComponents = structuredBody?.ContainsKey("component") is true ? structuredBody["component"] : null;
```

Comment:
Could use GetValueOrDefault() ?

Reply:
Here ```IDictionary``` only have ```bool TryGetValue(string key, out value)``` method, the ```GetValueOrDefault()``` belongs to ```Dictionary``` which may not the exact instance here.

Reply:
Do we need to transform it to IDictionary? Since element data["ClinicalDocument"] is a Dictionary.

Reply:
'IDictionary' is better here.

Reply:
From offline discussions, still keep 'IDictionary' here in case we internally use instances that are not dictionaries in the future.

Reply:
why not use TryGetValue here?

Reply:
The ```bool TryGetValue(TKey key, out TValue value)``` still need us to define some intermediate variable in ```out value``` parameter, I'll take define an IDictionary extension so the code here can be simplified.

Reply:
For ccda input data we parse it to `Dictionary` object https://github.com/microsoft/FHIR-Converter/blob/027f35519d72c5b08fd24c45b8c029958919716a/src/Microsoft.Health.Fhir.Liquid.Converter/DictionaryJsonConverter.cs#L85
 but `Idictionary` outside, do we need to make them consistent? https://github.com/microsoft/FHIR-Converter/blob/027f35519d72c5b08fd24c45b8c029958919716a/src/Microsoft.Health.Fhir.Liquid.Converter/Parsers/CcdaDataParser.cs#L46

If they are ccda specific filters, the input will always be `Dictionary` for now.

Reply:
Here we assume the data structure always be like ```data["ClinicalDocument"]["component"]["structuredBody"]["component"]```, which indicates the input must be **ccda** format.

I think the using ```Dictionary``` internally can be enough for now, if we update this ```SectionFilter``` to take more input format in the future, we can then use ```IDictionary``` instead.

Reply:
Have update to ```as Dictionary``` and use ```GetValueOrDefault```, same with other code in this filter.